python/summarize_sgfs.py

```diff
@@ -0,0 +1,120 @@
+from sgfmill import sgf
+import elo
+import os
+import argparse
+
+class SummarizeSGFs:
+  """Summarize Go games in sgfs file format. Each row of sgfs game is a single game in a sgf string."""
+
+  def __init__(
+      self,
+      file_path
+  ):
+    self.file_path = file_path
+    self.sgf_files = []
+    self.players = {}
+    self.results = []
```

Comment:
Why not have this be a dictionary?

You could have a collections.defaultdict mapping tuples of (black_player_name, white_player_name) -> WinLossStats, 

where WinLossStats could be a dataclass https://docs.python.org/3/library/dataclasses.html that just stores the count of wins and losses, defaulting to 0 on construction.

Then you could just do something like `self.resullts[(pla_black,pla_white)].wins += 1`.

This means you can get rid of all of the indexing logic, all of the table resizing code, and the player info helper, and it also becomes easy to enable the calculation of the P1Advantage since we keep track of who played  black and who played white.

You also don't need to store the set of players. If you set it up like this, you can compute it on the fly like:
`players = set(itertools.chain(*((pla_black,pla_white) for (pla_black,pla_white) in self.results.keys())))`

Reply:
Very good suggestion. Now it has changed to a dictionary, and also use `players = set(itertools.chain(*(pla_names for pla_names in self.results.keys())))` to get player information. Remove the indexing log and resizing codes.